torch/_functorch/_aot_autograd/schemas.py

```diff
@@ -623,6 +623,7 @@ class AOTConfig:
     aot_autograd_arg_pos_to_source: Optional[List[Source]] = None
     inference_compiler: Optional[Callable] = None
     enable_log: bool = True
+    scope_root: Optional[torch.nn.Module] = None
```

Comment:
I think it's unclear from the outside what `scope_root` is and what behavior it's supposed to control. There is no way of telling (without knowing the code) that `scope_root` controls whether `nn_module_stack` is populated.

I suggest we have the name more explicitly describe the behavior that setting this config option will produce, as well as adding some comments documenting it.

Reply:
We now instead just attach private attr on the orig module